src/ui/ActionFilters/McExceptionFilter.cs

```diff
@@ -3,41 +3,65 @@
 using Microsoft.ApplicationInsights;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.Extensions.Logging;
 
 namespace GovUk.Education.ManageCourses.Ui.ActionFilters
 {
     public class McExceptionFilter : IExceptionFilter
     {
+        private readonly ILogger _logger;
+
+        public McExceptionFilter(ILogger<McExceptionFilter> logger)
+        {
+            _logger = logger;
+        }
+
         public void OnException(ExceptionContext context)
         {
             var swaggerException = context.Exception as SwaggerException;
-            
-            if (swaggerException?.StatusCode >= 400 && swaggerException?.StatusCode <= 404)
+
+            if (swaggerException?.StatusCode >= 400 && swaggerException.StatusCode <= 404)
             {
-                context.Result = new RedirectToActionResult("Index", "Error", new { statusCode = swaggerException?.StatusCode });
+                context.Result = StatusCodeViewResult(context, swaggerException.StatusCode);
                 context.ExceptionHandled = true;
+                return;
             }
-            else if (swaggerException?.StatusCode == (int) HttpStatusCode.UnavailableForLegalReasons)
+
+            if (swaggerException?.StatusCode == (int)HttpStatusCode.UnavailableForLegalReasons)
             {
                 context.Result = new RedirectToActionResult("AcceptTerms", "Home", null);
                 context.ExceptionHandled = true;
+                return;
             }
-            else if (swaggerException != null)
-            {                    
-                context.Result = new RedirectToActionResult("Index", "Error", new { statusCode = 500 });
-                context.ExceptionHandled = true;
-            }
-            else
+
+            if (swaggerException == null)
             {
                 // not a swagger exception - so make sure AI gets notified despite us handling it here.
 
                 // MS Doc examples instantiate TelemetryClient immediately before use rather than keeping an
                 // instance around. It's not clear why but here I'm following this example.
                 new TelemetryClient().TrackException(context.Exception);
-                
-                context.Result = new RedirectToActionResult("Index", "Error", new { statusCode = 500 });
-                context.ExceptionHandled = true;
             }
+
+            _logger.LogError(context.Exception, "Unhandled exception");
```

Comment:
Can we also log url they came from and/or url they going to?

Reply:
as well as if the user is logged in and there subject id

Reply:
It looks like application insights is already gathering the extra context about URLs. Not sure about current user info. Please card that up as that's beyond the objective of this change.

Reply:
https://trello.com/c/Bj4sju1N/304-logging-for-mcexceptionfiltercs-manage-course-ui